### PR TITLE
Adds support for .NET Framework 4.5 in the native loader.

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR adds support for .NET Framework 4.5 in the native loader (proxy).

@DataDog/apm-dotnet

### CODE REVIEW

File: tracer/src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp [hunk 1]
```diff
@@ -180,7 +197,15 @@ namespace datadog::shared::nativeloader
...
                 DWORD local_mask_hi;
-                HRESULT hr = info5->GetEventMask2(&local_mask_low, &local_mask_hi);
+                HRESULT hr;
+                if (info5 != nullptr)
+                {
+                    hr = info5->GetEventMask2(&local_mask_low, &local_mask_hi);
+                }
+                else
+                {
+                    hr = info4->GetEventMask(&local_mask_low);
```
[the remark this answers is not in the document]
Comment (tonyredondo) [reply tracer/src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp:207]: done in 11f8091

**Review by gleocadie — APPROVED** [review]
LGTM

**Review by zacharycmontoya — APPROVED** [review]
LGTM

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #1825 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/a2561d2153cb020c651a1c87595be50b1ddfe2a1) will **not change** line coverage
:heavy_check_mark: Merging #1825 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/a2561d2153cb020c651a1c87595be50b1ddfe2a1) will **not change** branch coverage
:no_entry: Merging #1825 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/a2561d2153cb020c651a1c87595be50b1ddfe2a1) will will **increase** complexity by `6`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/a2561d2153cb020c651a1c87595be50b1ddfe2a1) | #1825       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9431` / `15040` | `9441` / `15052` |          |
| Lines %   | `63%`      | `63%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4158` / `6522` | `4168` / `6528` |          |
| Branches %| `64%`      | `64%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7684`      | `7690`        |  `6`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=76128&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/76128/artifacts?artifactName=Code%20Coverage%20Report_76128&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1825](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=76197&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/76197/artifacts?artifactName=Code%20Coverage%20Report_76197&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/a2561d2153cb020c651a1c87595be50b1ddfe2a1) | #1825       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `63%`      | `63%`       |  `0%` :heavy_check_mark:  |
| Branches %| `64%`      | `64%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7684`      | `7690`       |  `6` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations.TaskContinuationGenerator`4](https://github.com/DataDog/dd-trace-dotnet/tree/99a6575105576858caa54cbd27fe5b0109fcd17f/tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/Continuations/TaskContinuationGenerator`1.cs) | `-5%` :no_entry: | `-9%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.RuntimeMetrics.RuntimeMetricsWriter](https://github.com/DataDog/dd-trace-dotnet/tree/99a6575105576858caa54cbd27fe5b0109fcd17f/tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs) | `0%` :heavy_check_mark: | `8%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.LogRateLimiter](https://github.com/DataDog/dd-trace-dotnet/tree/99a6575105576858caa54cbd27fe5b0109fcd17f/tracer/src/Datadog.Trace/Logging/LogRateLimiter.cs) | `0%` :heavy_check_mark: | `17%` :heavy_check_mark:   | `0` :heavy_check_mark: |


The following classes were added in #1825:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Elasticsearch.V7.ElasticsearchV7Constants](https://github.com/DataDog/dd-trace-dotnet/pull/1825/files) | `100%` | `100%` | `1` |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Elasticsearch.V7.Transport_Request_Integration](https://github.com/DataDog/dd-trace-dotnet/pull/1825/files) | `100%` | `70%` | `11` |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Elasticsearch.V7.Transport_RequestAsync_Integration](https://github.com/DataDog/dd-trace-dotnet/pull/1825/files) | `100%` | `70%` | `11` |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.ValidateAsync4Integration](https://github.com/DataDog/dd-trace-dotnet/pull/1825/files) | `82%` | `33%` | `7` |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.ValidateAsyncIntegration](https://github.com/DataDog/dd-trace-dotnet/pull/1825/files) | `82%` | `33%` | `7` |


5 classes were removed from Datadog.Trace in #1825


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=76128&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/76128/artifacts?artifactName=Code%20Coverage%20Report_76128&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1825](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=76197&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/76197/artifacts?artifactName=Code%20Coverage%20Report_76197&api-version=6.0&%24format=zip)

**Pull request merged**